### ISSUE TITLE
ParalleIOTest failing

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#17]
@C0nsultant currently, the `ParalleIOTest.cpp` does not compile.

Issues are:
- access to the `Series()` object
- changed API of the `loadChunk()` member function.

I think the first issue is solvable via
```diff
diff --git a/test/ParallelIOTest.cpp b/test/ParallelIOTest.cpp
index cef531f..b45c9e6 100755
--- a/test/ParallelIOTest.cpp
+++ b/test/ParallelIOTest.cpp
@@ -35,9 +35,9 @@ BOOST_AUTO_TEST_CASE(git_hdf5_sample_content_test)
     MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
     /* only a 3x3x3 chunk of the actual data is hardcoded. every worker reads 1/3 */
     uint64_t rank = mpi_rank % 3;
-    Series o = Series("../samples/git-sample/",
-                      "data00000%T.h5",
-                      true);
+    Series o = Series::read("../samples/git-sample/",
+                            "data00000%T.h5",
+                            true);
 
 
     {
```

Can you please provide a PR that fixes the test?

**Comment by ax3l** [comment]
I will just disable the test in  #13 for now so I can finish my contribution.
Please feel free to comment in the one line in `CMakeLists.txt` that disables it as soon as this test is fixed:
```cmake
#    openPMDParallelIO # TODO: Broken! Does not compile
```

**Comment by ax3l** [comment]
compiles now but I still get a RT segfault on travis: https://travis-ci.org/ComputationalRadiationPhysics/libopenPMD/jobs/337479031

**Comment by anokfireball** [comment]
Reducing this to the minimal faulty part (Boost test + MPI strips the error messages)
```cpp
#include "Series.hpp"

int main(int argc, char *argv[])
{
    MPI::Init(argc, argv);
    Series o = Series::read("../samples/git-sample/data00000%T.h5", MPI_COMM_WORLD);
    MPI::Finalize();
    return 0;
}
```
reveals that tearing down the user-created HDF5 objects and opened files fails:

> HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5P.c line 1517 in H5Pclose(): can't close
    major: Property lists
    minor: Unable to free object
  #001: H5I.c line 1352 in H5I_dec_app_ref(): can't decrement ID ref count
    major: Object atom
    minor: Unable to decrement reference count
  #002: H5I.c line 1282 in H5I_dec_ref(): can't locate ID
    major: Object atom
    minor: Unable to find atom information (already closed?)
Interal error: Failed to close HDF5 dataset transfer property
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5P.c line 1517 in H5Pclose(): can't close
    major: Property lists
    minor: Unable to free object
  #001: H5I.c line 1352 in H5I_dec_app_ref(): can't decrement ID ref count
    major: Object atom
    minor: Unable to decrement reference count
  #002: H5I.c line 1282 in H5I_dec_ref(): can't locate ID
    major: Object atom
    minor: Unable to find atom information (already closed?)
Interal error: Failed to close HDF5 file access property
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5T.c line 1721 in H5Tclose(): not a datatype
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 enum
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5F.c line 769 in H5Fclose(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 file
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5F.c line 769 in H5Fclose(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 file
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5F.c line 769 in H5Fclose(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 file
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5F.c line 769 in H5Fclose(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 file
HDF5-DIAG: Error detected in HDF5 (1.10.1) thread 0:
  #000: H5F.c line 769 in H5Fclose(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
Internal error: Failed to close HDF5 file

Considering the official documentation for [```H5Fclose```](https://support.hdfgroup.org/HDF5/doc/RM/RM_H5F.html#File-Close), specifically
> For example, an MPI-IO file close is a collective call; [...]
> In such situations, applications must close all open objects in a file before calling H5Fclose.

this most likely means that not all HDF5 objects are closed before calling H5Fclose in the parallel context.

**Comment by anokfireball** [comment]
This was fixed in #30 .
The problem was the scope of the Series. ```MPI_Finalize``` was called *before* the destructor of an MPI-aware HDF5 context had completed, resulting in UB.
